ci/README.md

```diff
@@ -0,0 +1,111 @@
+
+The nodesets select the hosts based on labels in zuul.
+The labels available in zuul are shown on the [RDO Zuul labels tab](https://review.rdoproject.org/zuul/labels).
+
+The labels used for the nodesets are `coreos-crc-extracted-<crc_version>-<size>`.
+The “extracted” CRC describes the way that the job deploys and interacts with CRC.
+
+Usually, CRC is run using the `crc start` command, which created a VM on your host which runs the OCP cloud.
+In Zuul, the provisioned hosts are also virtual machines, so running `crc start` would result in a VM in a VM. This nested virtualisation causes some performance issues.
+
+The `extracted` deployment try to address the performance issues associated with nested virtualisation. The infrastructure is more complicated than nested.
+The `coreos-crc-extracted-...` labels provide a VM with an extracted CRC VM image, so that the CRC VM can be booted directly by the cloud provider. The `crc` VM is not accessed directly, but via a second `controller` VM, on which tests are run. The `stf-base-2node` job includes a network configuration to make sure the controller can communicate with the OCP deployment in CRC. All ansible playbooks should be run against `controller`. The `stf-base-2node` job is a common parent to all the jobs in the repo, and should rarely need an update.
+
+The name of each nodeset corresponds to the version of OCP that is deployed by the CRC image.
+
+## Adding new jobs
+
+If a new job needs to be added, it should inherit from `stf-base` ( or one of its child-jobs) which includes common tasks for setting up STF. The new jobs should have minimal configuration lines; either the `scenario` var is passed, which selects a vars file for stf-run-ci, to change its configuration, or the nodeset should be updated, which selects the OCP version.
+
+Below are examples of how to add a job. Take note of how the `scenario` var and the `nodeset` is passed.
+
+    - job:
+        name: stf-crc-nightly_bundles
+        parent: stf-base
+        abstract: true
+        description: |
+          Example of a job that extends the stf-base-job, and passes the "nightly_bundles" scenario var. This job does NOT have a nodeset defined so it must be abstract.
+        vars:
+          scenario: "nightly_bundles"
+
+OR
+
+    - job:
+        name: stf-crc-ocp_414-nightly_bundles
+        parent: stf-crc-nightly_bundles
+        description: |
+          Example of a job defining a nodeset to be used.
+          Since this job derives from a job with a scenario, it can be run directly.
+        nodeset: stf-crc_extracted-ocp414
+
+All non-abstract jobs inheriting from `stf-base` must pass a `scenario` var to work correctly. There is no default value for the `scenario`.
+All non-abstract jobs defined in this repo must have a `nodeset` to run correctly. Specifically, the nodeset must include nodes called `controller` and `crc`. This requirements comes from the `stf-base-2node` job.
+
```

Comment:
We need to mention that after define the job it should be added to the project template or under the project itself to be fired.